Mathlib/GroupTheory/GroupAction/SubMulAction.lean

```diff
@@ -0,0 +1,386 @@
+/-
+Copyright (c) 2020 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+
+! This file was ported from Lean 3 source module group_theory.group_action.sub_mul_action
+! leanprover-community/mathlib commit f93c11933efbc3c2f0299e47b8ff83e9b539cbf6
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Hom.GroupAction
+import Mathlib.Algebra.Module.Basic
+import Mathlib.Data.SetLike.Basic
+import Mathlib.GroupTheory.GroupAction.Basic
+
+/-!
+
+# Sets invariant to a `MulAction`
+
+In this file we define `SubMulAction R M`; a subset of a `MulAction R M` which is closed with
+respect to scalar multiplication.
+
+For most uses, typically `Submodule R M` is more powerful.
+
+## Main definitions
+
+* `SubMulAction.mulAction` - the `MulAction R M` transferred to the subtype.
+* `SubMulAction.mulAction'` - the `MulAction S M` transferred to the subtype when
+  `IsScalarTower S R M`.
+* `SubMulAction.isScalarTower` - the `IsScalarTower S R M` transferred to the subtype.
+
+## Tags
+
+submodule, mul_action
+-/
+
+
+open Function
+
+universe u u' u'' v
+
+variable {S : Type u'} {T : Type u''} {R : Type u} {M : Type v}
+
+/-- `SMulMemClass S R M` says `S` is a type of subsets `s ≤ M` that are closed under the
+scalar action of `R` on `M`. -/
+class SMulMemClass (S : Type _) (R M : outParam <| Type _) [SMul R M] [SetLike S M] where
```

Comment:
Could you for all these classes please remove the `outParam` annotations on `M` (keeping them for `R`), following #1832?

```suggestion
class SMulMemClass (S : Type _) (R : outParam <| Type _) (M : Type _) [SMul R M] [SetLike S M] where
```